File: app/models/shared_resource.rb
class SharedResource < ApplicationRecord
  belongs_to :user
  belongs_to :shareable, polymorphic: true

  enum permission_level: { edit: "edit", full_access: "full_access",  view: "view" }

  validates :permission_level, presence: true
  validates :user_id, presence: true
end

File: app/javascript/app/models/SharedResource.ts
export type SharedResource = {
  id?: number;
  shareable_type?: "Project" | "Document" | "Report";
  shareable_id?: number;
  user_id: number;
  permission_level: 'full access' | 'edit' | 'view';
  created_at?: string;
  updated_at?: string;
}

  